### DISCUSSION TITLE
Help building a local site with relativeURL

### DISCUSSION THREAD

**Discussion by primercuervo** [nunocoracao/blowfish#1140] [Q&A]
### Describe the scenario
I would like to create a local website for display using the blowfish template. According to the [docs about hosting](https://blowfish.page/docs/hosting-deployment/)
> Blowfish is built using relative URLs throughout the theme. This enables sites to easily be deployed to sub-folders and hosts like GitHub Pages. There’s usually no special configuration required for this to work as long as the baseURL parameter has been configured in the config.toml file.

So I tried doing so, but the `index.html` doesn't show as expected.

### Steps to reproduce
1. Clone the [blowfish template](https://github.com/nunocoracao/blowfish_template) repository
2. set up the theme with `git submodule update --init`
3. Add the following two parameters to `{REPO}/config/_default/config.toml`
`relativeURLs = true`
`uglyURLs = true`
4. build the website with the `hugo` command

### Expected result
I expected to see the website just as it shows with `hugo server --disableFastRender`
![image](https://github.com/nunocoracao/blowfish/assets/12393769/a59419fd-4c61-4583-83ef-bbd7955a7d5d)

### Actual result
the CSS is being blocked by the browsers. Tried in both Chrome and Firefox:
![image](https://github.com/nunocoracao/blowfish/assets/12393769/0aa192ab-e3e5-4ea3-978d-52a5d2bd7b1b)

Following errors are showing up in the console:
>“file:///home/cuervo/git/blowfish_template/public/css/main.bundle.min.2b8b793b5350f23c027e845b237b267c85c08a4a6128dfad2b43b25f354c42eb4629bb7f88b75e09b52056c28e0d057594a8185bd540bd7bece42b3be6399cdf.css” is not eligible for integrity checks since it’s neither CORS-enabled nor same-origin. [index.html](file:///home/cuervo/git/blowfish_template/public/index.html)
“file:///home/cuervo/git/blowfish_template/public/js/main.bundle.min.1271c22a557a15f10b5d38d2c87e49503604cfeffebadc1e7c267934e4b9f3ba1b74d81371219f9681dab2141482fc4ed2a8b462b179f0ef43a5f76d0575e85b.js” is not eligible for integrity checks since it’s neither CORS-enabled nor same-origin. [index.html](file:///home/cuervo/git/blowfish_template/public/index.html)
“file:///home/cuervo/git/blowfish_template/public/js/process.min.62060bb247f4de2b6dde45903668fefb68d792f365587605177b1227c0cf43588701edaca0cb40e2c8e2789bd5ce67c1d2a215b9fb258c3496a7cd25e7cb5fdf.js” is not eligible for integrity checks since it’s neither CORS-enabled nor same-origin. [index.html](file:///home/cuervo/git/blowfish_template/public/index.html)
“file:///home/cuervo/git/blowfish_template/public/js/appearance.min.516a16745bea5a9bd011138d254cc0fd3973cd55ce6e15f3dec763e7c7c2c7448f8fe7b54cca811cb821b0c7e12cd161caace1dd794ac3d34d40937cbcc9ee12.js” is not eligible for integrity checks since it’s neither CORS-enabled nor same-origin. [index.html](file:///home/cuervo/git/blowfish_template/public/index.html)
“file:///home/cuervo/git/blowfish_template/public/js/main.bundle.min.1271c22a557a15f10b5d38d2c87e49503604cfeffebadc1e7c267934e4b9f3ba1b74d81371219f9681dab2141482fc4ed2a8b462b179f0ef43a5f76d0575e85b.js” is not eligible for integrity checks since it’s neither CORS-enabled nor same-origin. [index.html](file:///home/cuervo/git/blowfish_template/public/index.html)
“file:///home/cuervo/git/blowfish_template/public/js/process.min.62060bb247f4de2b6dde45903668fefb68d792f365587605177b1227c0cf43588701edaca0cb40e2c8e2789bd5ce67c1d2a215b9fb258c3496a7cd25e7cb5fdf.js” is not eligible for integrity checks since it’s neither CORS-enabled nor same-origin.

Unfortunately, I'm not versed on website development, and thought of setting this kind of page for a familiar project. The result I get with `hugo serve` fulfills all our expectations (thanks a lot for this template!), but we're limited in knowledge how to make the local page work without a server command. I have been searching online for a solution but haven't found a consistent answer. Some suggest that I modify the template to remove all integrity checks, but I'm not certain this is recommended.

Any help would be highly appreciated. 

hugo version: v0.121.1-00b46fed8e47f7bb0a85d7cfc2d9f1356379b740+extended linux/amd64 BuildDate=2023-12-08T08:47:45Z VendorInfo=gohugoio
Blowfish version: v2.48.0

**Comment by primercuervo** [comment]
@nunocoracao do you perhaps have any suggestions about this usecase?

**Reply by demodeveloperlab** [reply]
Go step by step
[Build your page using Blowfish](https://n9o.xyz/posts/202310-blowfish-tutorial/)

Faust

**Reply by primercuervo** [reply]
@demodeveloperlab that site does not answer my question. My question is basically how to run the site, serverless-ly. Without hosting online, and without having a local server (if possible). This is to run the site on a machine not connected to the internet. Currently, I run a local http-server that points to the `index.html`. 

But I wonder if this step of running the http-server can be skipped by configuring the theme so that no integrity checks are required.
